proposals/expand-ref.md

```diff
@@ -0,0 +1,254 @@
+
+```csharp
+ref struct Deserializer
+{
+    ref scoped Utf8JsonReader reader;
+
+    ReadOnlySpan<byte> M1()
+    {
+        // okay: implicitly scoped to current method
+        var span = reader.ValueSpan; 
+
+        // okay
+        reader.Skip();
+
+        // Error: can't escape the ref data the ref scoped field refers to
+        return reader.ValueSpan;
+    }
+}
+```
+
+This is accomplished by giving every `ref scoped` field a new two new escape scopes named _current field N_ and _current ref field N_ where _N_ is the numeric order of the field. For example the first field has a _safe-to-escape_ of _current field 1_ and a _ref-safe-to-escape_ of _current ref field N_. Both escape scopes can be converted to _current method_, and _current field N_ can be converted to _current ref field N_, but no other defined relationships exist. That serves to restrict their usage to the current method where the containing value is used. This escape scope applies to both 
+
+The [method arguments must match](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/low-level-struct-improvements.md#rules-method-arguments-must-match) rules do not need to be updated here as they already account for `ref` parameters being captured as `ref` field. Even though a `ref` to `ref struct` was not directly returnable before, it could be returned indirectly by a `ref` to a `struct` field of the value.
+
+The language will also allow for `ref` fields to be declared as `scoped ref`. There are less use cases for this but `ref scoped` implies `scoped ref` hence the rules must be adjusted to account for this. As such the syntax will be exposed because while the use cases are small the infrastructure already exists. The _ref-safe-to-escape_ of such fields follows the logic above for `ref scoped` fields.
+
+Detailed notes:
+- A `ref` field where the type is a `ref struct` must be `ref scoped`
+- A `ref` field may be marked `scoped ref`
+
+### Sunset restricted types
+The ability for any type to be a `ref` field allows us to fully sunset the notion of restricted types.  The compiler has a concept of a set of _restricted types_ which is largely undocumented. These types were given a special status because in C# 1.0 there was no general purpose way to express their behavior. Most notably the fact that the types can contain references to the execution stack. Instead the compiler had special knowledge of them and restricted their use to ways that would always be safe: disallowed returns, cannot use as array elements, cannot use in generics, etc ...
+
+Once `ref` fields are available and extended to support `ref struct` these types can be fully rationalized within those rules. As such the compiler will no longer have the notion of restricted types when using a language version that supports `ref` fields of `ref struct`.
+
+To support this our `ref` safety rules will be updated as follows:
```

Comment:
Can we assume updates to the following don't indicate anything other than clarity on their semantics rather than changing anything about their "officially supported" state.

Reply:
Correct. This is just about getting clarity on the semantics. Particularly I know the runtime team has wanted to leverage `TypedReference` more and this would provide clarity on what can and cannot be done safely.